### ISSUE TITLE
Plugin crashes if DHCP network ID is nil

### ISSUE THREAD

**Issue by Shilpa-Gokul** [IBM-Cloud/terraform-provider-ibm#4683]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* ibm_pi_dhcps

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
data "ibm_pi_dhcps" "dhcp_services" {
  pi_cloud_instance_id =  <cloud_instance_id>
}
```

### Panic Output
https://gist.github.com/Shilpa-Gokul/69c37444decdaeddfe5073caccbe42fa

### Expected Behavior
Panic should not have occurred instead a graceful cleanup or shutdown should have happened

### Actual Behavior
As the network ID is nil, a panic occurred stating "invalid memory address or nil pointer dereference"

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by yussufsh** [comment]
Above PR will prevent plugin panic. The real issue is missing Network ID which needs to be checked by the service broker team.

**Comment by rmoralesjr** [comment]
On a dal10 staging, I see a nulls in network structure when in BUILD state. 
[{"id":"e58bdd7b-87c6-48cc-8a98-edc9264763","network":{"id":null,"name":null},"status":"BUILD"}]

Later the get DHCP returns the network info when in ACTIVE state.
[{"id":"e58bdd7b-87c6-48cc-8a98-edc9264763","network":{"id":"8710f1fc-6e7c-4482-b134-6695f65326","name":"DHCPSERVERrene-dal10-per_Private"},"status":"ACTIVE"}]

So the terraform code should be changed to handle this temporary condition. I am not sure if this is the root cause for this issue (until we get region zone info to locate logs) but at least we know one of the scenarios to reproduce this condition.

**Comment by yussufsh** [comment]
> On a dal10 staging, I see a nulls in network structure when in BUILD state. [{"id":"e58bdd7b-87c6-48cc-8a98-edc9264763","network":{"id":null,"name":null},"status":"BUILD"}]
> 
> Later the get DHCP returns the network info when in ACTIVE state. [{"id":"e58bdd7b-87c6-48cc-8a98-edc9264763","network":{"id":"8710f1fc-6e7c-4482-b134-6695f65326","name":"DHCPSERVERrene-dal10-per_Private"},"status":"ACTIVE"}]
> 
> So the terraform code should be changed to handle this temporary condition. I am not sure if this is the root cause for this issue (until we get region zone info to locate logs) but at least we know one of the scenarios to reproduce this condition.

When in build state dhcp.network should be nil or empty struct from the service broker instead of setting id and name to `null`.
The temporary fix is provided by #4685 which will also check if id is nil.
